packages/types/src/api-reference/api-reference-configuration.ts

```diff
@@ -0,0 +1,297 @@
+})
+
+/** Configuration for path-based routing */
+const PathRoutingSchema = z.object({
+  /** Base path for the API reference */
+  basePath: z.string(),
+})
+
+export const ApiReferenceConfigurationSchema = z
+  .object({
+    /** A string to use one of the color presets */
+    theme: ThemeIdEnum.optional().catch('default'),
+    /** The layout to use for the references */
+    layout: z.enum(['modern', 'classic']).optional().default('modern').catch('modern'),
+    /** The Swagger/OpenAPI spec to render */
+    spec: SpecConfigurationSchema.optional(),
+    /**
+     * URL to a request proxy for the API client
+     * @deprecated Use proxyUrl instead
+     */
+    proxy: z.string().optional(),
+    /** URL to a request proxy for the API client */
+    proxyUrl: z.string().optional(),
+    /**
+    Whether the spec input should show
+    @default false
+  */
+    isEditable: z.boolean().optional().default(false).catch(false),
+    /**
+    Whether to show the sidebar
+    @default true
+  */
+    showSidebar: z.boolean().optional().default(true).catch(true),
+    /**
+     * Whether to show models in the sidebar, search, and content.
+     * @default false
+     */
+    hideModels: z.boolean().optional().default(false).catch(false),
+    /**
+     * Whether to show the "Download OpenAPI Document" button
+     * @default false
+     */
+    hideDownloadButton: z.boolean().optional().default(false).catch(false),
+    /**
+     * Whether to show the "Test Request" button
+     * @default false
+     */
+    hideTestRequestButton: z.boolean().optional().default(false).catch(false),
+    /**
+     * Whether to show the sidebar search bar
+     * @default false
+     */
+    hideSearch: z.boolean().optional().default(false).catch(false),
+    /** Whether dark mode is on or off initially (light mode) */
+    darkMode: z.boolean().optional(),
+    /** forceDarkModeState makes it always this state no matter what */
+    forceDarkModeState: z.enum(['dark', 'light']).optional(),
+    /** Whether to show the dark mode toggle */
+    hideDarkModeToggle: z.boolean().optional(),
+    /** Key used with CTRL/CMD to open the search modal (defaults to 'k' e.g. CMD+k) */
+    searchHotKey: SearchHotKeyEnum.optional(),
+    /**
+     * If used, passed data will be added to the HTML header
+     * @see https://unhead.unjs.io/usage/composables/use-seo-meta
+     */
+    metaData: z.any().optional(), // Using any for UseSeoMetaInput since it's an external type
+    /**
+     * Path to a favicon image
+     * @default undefined
+     * @example '/favicon.svg'
+     */
+    favicon: z.string().optional(),
+    /**
+     * List of httpsnippet clients to hide from the clients menu
+     * By default hides Unirest, pass `[]` to show all clients
+     */
+    hiddenClients: z.union([z.record(z.union([z.boolean(), z.array(z.string())])), z.array(z.string())]).optional(),
```

Comment:
Should we transform to the string array?

Reply:
Not in this pr